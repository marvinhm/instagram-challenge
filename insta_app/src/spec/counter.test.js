import React from 'react';
import Counter from '../components/counter';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('Counter component', () => {
    it('Should match the snapshot', () => {
        const tree = renderer.create(<Counter />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('Starts with a count of 0', () => {
        const wrapper = shallow(<Counter/>)
        const text = wrapper.find('p').text()
        // const countState = wrapper.state().count
        expect(text).toEqual("You clicked 0 times")
    })

    it('can increment the count when button clicked', () => {
        const wrapper = shallow(<Counter/>)
        const incrementBtn = wrapper.find('button.increment')
        incrementBtn.simulate('click')

        const text = wrapper.find('p').text()
        expect(text).toEqual("You clicked 1 times")
    })

    it('can decrement the count when button clicked', () => {
        const wrapper = shallow(<Counter/>)
        const decrementBtn = wrapper.find('button.decrement')
        decrementBtn.simulate('click')

        const text = wrapper.find('p').text()
        expect(text).toEqual("You clicked -1 times")
    })
    
})
