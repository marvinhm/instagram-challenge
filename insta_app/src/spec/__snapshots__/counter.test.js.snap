// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component Should match the snapshot 1`] = `
<div
  className="App"
>
  <p>
    You clicked 
    0
     times
  </p>
  <button
    className="increment"
    onClick={[Function]}
  >
    +++
  </button>
  <button
    className="decrement"
    onClick={[Function]}
  >
    ---
  </button>
</div>
`;
